x-pack/plugins/observability_solution/slo/public/application.tsx

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { EuiErrorBoundary } from '@elastic/eui';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { KibanaThemeProvider } from '@kbn/react-kibana-context-theme';
+import { AppMountParameters, APP_WRAPPER_CLASS, CoreStart } from '@kbn/core/public';
+import type { LazyObservabilityPageTemplateProps } from '@kbn/observability-shared-plugin/public';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { Router, Routes, Route } from '@kbn/shared-ux-router';
+import { EuiThemeProvider } from '@kbn/kibana-react-plugin/common';
+import { RedirectAppLinks } from '@kbn/shared-ux-link-redirect-app';
+import { Storage } from '@kbn/kibana-utils-plugin/public';
+import { ObservabilityRuleTypeRegistry } from '@kbn/observability-plugin/public';
+
+import { PluginContext } from './context/plugin_context';
+
+import { SloPublicPluginsStart } from './types';
+import { routes } from './routes/routes';
+
+function App() {
+  return (
+    <>
+      <Routes>
+        {Object.keys(routes).map((key) => {
+          const path = key as keyof typeof routes;
+          const { handler, exact } = routes[path];
+          const Wrapper = () => {
+            return handler();
+          };
+          return <Route key={path} path={path} exact={exact} component={Wrapper} />;
+        })}
+      </Routes>
+    </>
+  );
+}
+
+export const renderApp = ({
+  core,
+  plugins,
+  appMountParameters,
+  ObservabilityPageTemplate,
+  isDev,
+  kibanaVersion,
+  isServerless,
+  observabilityRuleTypeRegistry,
+}: {
+  core: CoreStart;
+  plugins: SloPublicPluginsStart;
+  appMountParameters: AppMountParameters;
+  observabilityRuleTypeRegistry: ObservabilityRuleTypeRegistry;
+  ObservabilityPageTemplate: React.ComponentType<LazyObservabilityPageTemplateProps>;
+  isDev?: boolean;
+  kibanaVersion: string;
+  isServerless?: boolean;
+}) => {
+  const { element, history, theme$ } = appMountParameters;
+  const i18nCore = core.i18n;
+  const isDarkMode = core.theme.getTheme().darkMode;
+
+  // ensure all divs are .kbnAppWrappers
+  element.classList.add(APP_WRAPPER_CLASS);
+
+  const queryClient = new QueryClient();
+  const CloudProvider = plugins.cloud?.CloudContextProvider ?? React.Fragment;
+
+  const PresentationContextProvider = plugins.presentationUtil?.ContextProvider ?? React.Fragment;
+
+  ReactDOM.render(
+    <PresentationContextProvider>
+      <EuiErrorBoundary>
+        <KibanaThemeProvider {...{ theme: { theme$ } }}>
+          <CloudProvider>
```

Comment:
Do we need CloudProvider here? I took it from current observability plugin and I want to know if it is redundant or not. I was checking the exploratory_view plugin, since it was also moved recently to its own plugin and I see CloudProvider is not used [there](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/exploratory_view/public/application/index.tsx)